source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -118,6 +127,27 @@ void generateV2LocalHeader(Buffer::Instance& out) {
   out.add(addr_fam_protocol_and_length, 4);
 }
 
+void generateV2Header(const Network::ProxyProtocolData& prox_proto_data, Buffer::Instance& out) {
+  uint16_t extension_length = 0;
+  for (auto& tlv : prox_proto_data.tlv_vector_) {
+    // TODO: handle overflow without crashing.
+    // assert(tlv.value.size() + 3u < std::numeric_limits<uint16_t>::max() - extension_length);
+    extension_length += 3 + tlv.value.size();
+  }
+  const auto& src = *prox_proto_data.src_addr_->ip();
+  const auto& dst = *prox_proto_data.dst_addr_->ip();
+  generateV2Header(src.addressAsString(), dst.addressAsString(), src.port(), dst.port(),
+                   src.version(), extension_length, out);
+
+  // Generate the TLV vector.
+  for (auto& tlv : prox_proto_data.tlv_vector_) {
+    out.add(&tlv.type, 1);
+    uint16_t size = htons(static_cast<uint16_t>(tlv.value.size()));
+    out.add(&size, 2);
```

Comment:
This is "2" because sizeof(uint16_t) is 2?

Reply:
Right. Updated to `sizeof()`.